compiler/src/lib.rs

```diff
@@ -36,17 +36,16 @@ pub fn compile_pil_or_asm<T: FieldElement>(
     output_dir: &Path,
     force_overwrite: bool,
     prove_with: Option<Backend>,
-) -> PathBuf {
+) -> Option<CompilationResult<T>> {
```

Comment:
Why Option?

Reply:
`compile_asm_string()` returns `None` if the ASM file has been compiled before. If you run `powdr pil -f`, you always get a result.